### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by sfkamath**
This commit consists of the dockerised reference code with variables 12 factored out. 

The impact to the existing code base and tomcat deploy scripts were in changes to external variables that allow it to differentiate between inter server (container to container) traffic and browser to server communication (over redirects). The browser to server communication is represented via variables with the EXT_ prefix to indicate 'external' to container traffic

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- Run the `build_docker_image.sh` script to build a docker image
+- Build the application with the `compile.sh` script if needed. This is needed if the reference code is modified or for example the eidas.xml or sp.properties file changes while setting up the environment variables
+- Start the application with docker
+- Start the docker containers
+  `docker-compose up -d`    starts the application as a daemon
+  `docker-compose logs -f`  tails the container logs
+- Stop the docker containers
+  `docker-compose down`     stops the containers and cleans up
+
+Alternately
```
Comment (adityapahuja) [inline docker/README.md]: Please replace 'Alternately' with 'Alternatively'.

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+      - "EXT_IDP_URL="
+      - "EXT_PROXY_URL="
+
+      - "COUNTRY_1_CODE=NL"
+      - "COUNTRY_1_NAME=The Netherlands"
+      - "COUNTRY_1_PROXY_URL=http://proxy-node-1:${DEFAULT_TOMCAT_PORT}"
+      - "COUNTRY_2_CODE=ES"
+      - "COUNTRY_2_NAME=Spain"
+      - "COUNTRY_2_PROXY_URL=http://proxy-node-2:${DEFAULT_TOMCAT_PORT}"
+      - "COUNTRY_3_CODE="
```
Comment (adityapahuja) [inline docker/docker-compose.yaml]: Do we need to have empty 3- 7 countries?

File: scripts/setenv/proxy-node-2.sh [hunk 3]
```diff
@@ -7,28 +7,34 @@ export EIDAS_CONFIG_REPOSITORY=./EIDAS-Config/
...
 
 # Shouldn't need this for the Proxy, but set it to empty to stop the variable replacement complaining:
-export SP_URL=''
 export COUNTRY_1_CODE=''
 export COUNTRY_1_NAME=''
 export COUNTRY_1_PROXY_URL=''
 export COUNTRY_2_CODE=''
 export COUNTRY_2_NAME=''
 export COUNTRY_2_PROXY_URL=''
+export COUNTRY_2_CODE=''
```
Comment (adityapahuja) [inline scripts/setenv/proxy-node-2.sh]: Please remove duplicate variables "COUNTRY_2".

File: docker/docker-compose.yaml (hunk 2 above)
Comment (sfkamath) [inline docker/docker-compose.yaml]: this reflects whats in the script files

File: docker/README.md (hunk 1 above)
Comment (sfkamath) [inline docker/README.md]: done

Comment (adityapahuja) [inline docker/README.md]: 👍

File: docker/docker-compose.yaml (hunk 2 above)
Comment (adityapahuja) [inline docker/docker-compose.yaml]: 👍

File: scripts/setenv/proxy-node-2.sh (hunk 3 above)
Comment (adityapahuja) [inline scripts/setenv/proxy-node-2.sh]: 👍

**Comment by adityapahuja** [comment]
Looks good to me.

**Pull request merged**